### PR TITLE
Fix rootblocks and eviction index 

### PR DESCRIPTION
**Pull request by jonastheis**
This PR fixes the eviction index of root blocks and adds a test case to make sure eviction and eviction delay of rootblocks is done correctly

### CODE REVIEW

File: pkg/protocol/engine/eviction/state_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package eviction
```
Comment (alexsporn) [inline pkg/protocol/engine/eviction/state_test.go:1]: `package eviction_test`

File: pkg/protocol/engine/eviction/testframework.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
+package eviction
```
Comment (alexsporn) [inline pkg/protocol/engine/eviction/testframework.go:1]: `package eviction_test` and add `_test.go` suffix if possible

File: pkg/model/eviction_index.go [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
+import (
+	iotago "github.com/iotaledger/iota.go/v4"
+)
 
 type EvictionIndex struct {
 	index *iotago.SlotIndex
 }
 
-func (e EvictionIndex) ShouldEvict(newIndex iotago.SlotIndex) bool {
+func NewEvictionIndex() *EvictionIndex {
```
Comment (alexsporn) [inline pkg/model/eviction_index.go:11]: do we need an empty `New` here or should the user just call `new(model.EvictionIndex)` ?

Comment (jonastheis) [reply pkg/model/eviction_index.go:11]: convention for almost all of our data types are they have a constructor. also we could add an optional initial index as a parameter to it

**Review by alexsporn — APPROVED** [review]

**Pull request merged**